library/std/src/sys/windows/alloc.rs

```diff
@@ -6,56 +8,81 @@ use crate::sys_common::alloc::{realloc_fallback, MIN_ALIGN};
 struct Header(*mut u8);
 
 unsafe fn get_header<'a>(ptr: *mut u8) -> &'a mut Header {
-    &mut *(ptr as *mut Header).offset(-1)
+    // SAFETY: the safety contract must be upheld by the caller
```

Comment:
What is the safety contract?

Reply:
Uhh, basically there's a Header located in the bytes directly before the pointer start?

Reply:
Ok, can you say that in the docs for the function?
```rust
/// # Safety
///
/// There must be a `Header` at `ptr.offset(-1)`.
```